src/gui_common/SegmentedBar.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A ProgressBar that is split up into IconProgressBars, data is stored in a dictionary
+/// </summary>
+public class SegmentedBar : ProgressBar
+{
+    public PackedScene IconProgressBarScene = GD.Load<PackedScene>("res://src/gui_common/IconProgressBar.tscn");
+    public string Type;
```

Comment:
Why are these public? Things that shouldn't be touched from elsewhere should be private by default.

Reply:
commenthing on this again.